packages/puppeteer/src/axePuppeteer.ts

```diff
@@ -23,40 +23,54 @@ function injectJSModule(frame: Frame): Promise<void> {
   });
 }
 
-function injectJSSource(frame: Frame, source: string | Function, args: any[] = []): Promise<void> {
+function injectJSSource(
+  frame: Frame,
+  source: string | Function,
+  args: any[] = []
+): Promise<void> {
   return frame.evaluate(source as any, ...args);
 }
 
-async function injectJS(frame: Frame, {source, selector, logOnError, args}: IInjectAxeArgs): Promise<void> {
-  const frames = await frame.$$(selector);
-  const injections = [];
-  for (const frameElement of frames) {
-    const subFrame = await frameElement.contentFrame();
-    const p = injectJS(subFrame as Frame, { source, selector, args, logOnError: true});
-    injections.push(p);
-  }
+async function injectJS(
+  frame: Frame,
+  { source, selector, logOnError, args }: IInjectAxeArgs
+): Promise<void> {
+  if (frame) {
```

Comment:
We should return early instead.

```suggestion
  if (!frame) {
    return;
  }
```